pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go

```diff
@@ -0,0 +1,145 @@
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
+)
+
+func TestMachineConfigReconciler(t *testing.T) {
+	fakeDh := func(controller *gomock.Controller) *mock_dynamichelper.MockInterface {
+		return mock_dynamichelper.NewMockInterface(controller)
+	}
+	cluster := func(enabled bool) *arov1alpha1.Cluster {
+		return &arov1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+			Status:     arov1alpha1.ClusterStatus{},
+			Spec: arov1alpha1.ClusterSpec{
+				OperatorFlags: arov1alpha1.OperatorFlags{
+					controllerEnabled: strconv.FormatBool(enabled),
+				},
+			},
+		}
+	}
+
+	t.Run("when no cluster resource is present, returns error", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		client := ctrlfake.NewClientBuilder().Build()
+		dh := fakeDh(controller)
+
+		r := NewMachineConfigReconciler(
+			logrus.NewEntry(logrus.StandardLogger()),
+			client,
+			dh,
+		)
+
+		request := ctrl.Request{}
+
+		_, err := r.Reconcile(context.Background(), request)
+
+		if !kerrors.IsNotFound(err) {
+			t.Errorf("wanted error: cluster not found, got error: %v", err)
+		}
+	})
+
+	t.Run("when controller is disabled, returns with no error", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		client := ctrlfake.NewClientBuilder().WithObjects(cluster(false)).Build()
+		dh := fakeDh(controller)
+
+		r := NewMachineConfigReconciler(
+			logrus.NewEntry(logrus.StandardLogger()),
+			client,
+			dh,
+		)
+
+		_, err := r.Reconcile(context.Background(), ctrl.Request{})
+
+		if err != nil {
+			t.Errorf("wanted no error, got error: %v", err)
+		}
+	})
+
+	t.Run("when no corresponding MachineConfigPool for the requested MachineConfig is present, does nothing", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		client := ctrlfake.NewClientBuilder().WithObjects(cluster(true)).Build()
+		dh := fakeDh(controller)
+
+		request := ctrl.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: "",
+				Name:      "99-custom-aro-dns",
+			},
+		}
+
+		r := NewMachineConfigReconciler(
+			logrus.NewEntry(logrus.StandardLogger()),
+			client,
+			dh,
+		)
+
+		_, err := r.Reconcile(context.Background(), request)
+
+		if err != nil {
+			t.Errorf("wanted no error, got error: %v", err)
+		}
+	})
+
+	t.Run("when a valid MachineConfigPool exists for the requested MachineConfig, reconciles MachineConifg", func(t *testing.T) {
+		controller := gomock.NewController(t)
+		defer controller.Finish()
+
+		client := ctrlfake.NewClientBuilder().
+			WithObjects(
+				cluster(true),
+				&mcv1.MachineConfigPool{
+					ObjectMeta: metav1.ObjectMeta{Name: "custom"},
+					Status:     mcv1.MachineConfigPoolStatus{},
+					Spec:       mcv1.MachineConfigPoolSpec{},
+				},
+			).
+			Build()
+		dh := fakeDh(controller)
+		dh.EXPECT().Ensure(gomock.Any(), gomock.Any()).Times(1)
```

Comment:
Let's leverage similar logic as here: https://github.com/Azure/ARO-RP/pull/2476/files?diff=unified&w=1#diff-71aa82188a09b613595179318fb771896687ffd693acc5187521d9586175b05eR115

```suggestion
		dh.EXPECT().Ensure(gomock.Any(), gomock.AssignableToTypeOf(&mcv1.MachineConfig{})).Times(1)

Reply:
Done